sunpy/map/tests/test_mapbase.py

```diff
@@ -1546,6 +1546,14 @@ def test_rotation_rect_pixelated_data(aia171_test_map):
     rect_rot_map = rect_map.rotate(30 * u.deg)
     rect_rot_map.peek()
 
+@figure_test
+def test_draw_contours_with_transform(aia171_test_map):
```

Comment:
Can you change this test to do the following:

A two-panel figure test, where the first panel is what is already written, and the second one composites in a scaling transform (e.g., by a factor of 2) like adding in https://matplotlib.org/stable/api/transformations.html#matplotlib.transforms.Affine2D

The contours will look "wrong" on the second panel, but the purpose is to confirm that they are "wrong" in the correct fashion.

Reply:
How does this look:

```python
import matplotlib.pyplot as plt
import astropy.units as u
import sunpy.map
from matplotlib.transforms import Affine2D

aia_map = sunpy.map.Map(AIA_171_IMAGE)
fig, axes = plt.subplots(1, 2, figsize=(16, 4))

for ax in axes:
    ax.set_axis_off()

# Panel 1: Original contours
ax1 = axes[0]
ax1 = fig.add_subplot(1, 2, 1, projection=aia_map)
aia_map.plot(axes=ax1, clip_interval=(1, 99.99)*u.percent)
aia_map.draw_contours([-1, 1]*u.percent, transform=ax1.get_transform('pixel'))

# Panel 2: Contours with scaling transform
ax2 = axes[1]
ax2 = fig.add_subplot(1, 2, 2, projection=aia_map)
scaling_transform = Affine2D().scale(2, 2)
composite_transform = ax2.get_transform('pixel') + scaling_transform
aia_map.plot(axes=ax2, clip_interval=(1, 99.99)*u.percent)
aia_map.draw_contours([-1, 1]*u.percent, transform=composite_transform)

# # Change x and y limits by a factor of two
# ax2.set_xlim(ax2.get_xlim()[0]*3, ax2.get_xlim()[1]*3)
# ax2.set_ylim(ax2.get_ylim()[0]*3, ax2.get_ylim()[1]*3)

ax2.set_title('Contours with Scaling Transform')

plt.show()
```
Produces:
<img width="976" alt="Screenshot 2024-02-02 at 2 13 17 PM" src="https://github.com/sunpy/sunpy/assets/48038715/769f9a23-f581-441e-a248-c13c6289386a">

If this looks good I will update the tests accordingly.

Reply:
Note that you can't actually see the contours at all on the right-hand panel.  That's because the order of the summation for the composite transform is wrong.  Here's a fixed version of your script (with the plotting code cleaned up as well):
```python
import matplotlib.pyplot as plt
from matplotlib.transforms import Affine2D

import astropy.units as u

import sunpy.map
from sunpy.data.sample import AIA_171_IMAGE

aia_map = sunpy.map.Map(AIA_171_IMAGE)
fig = plt.figure(figsize=(11, 4))

# Panel 1: Original contours
ax1 = fig.add_subplot(1, 2, 1, projection=aia_map)
aia_map.plot(axes=ax1, clip_interval=(1, 99.99)*u.percent)
aia_map.draw_contours([-1, 1]*u.percent, transform=ax1.get_transform('pixel'))

# Panel 2: Contours with scaling transform
ax2 = fig.add_subplot(1, 2, 2, projection=aia_map)
scaling_transform = Affine2D().scale(2, 2)
composite_transform = scaling_transform + ax2.get_transform('pixel')
aia_map.plot(axes=ax2, clip_interval=(1, 99.99)*u.percent)
aia_map.draw_contours([-1, 1]*u.percent, transform=composite_transform)

# # Change x and y limits by a factor of two
# ax2.set_xlim(ax2.get_xlim()[0]*3, ax2.get_xlim()[1]*3)
# ax2.set_ylim(ax2.get_ylim()[0]*3, ax2.get_ylim()[1]*3)

ax2.set_title('Contours with Scaling Transform')

plt.show()
```
![Figure_1](https://github.com/sunpy/sunpy/assets/991759/7701def5-17cd-4dc4-bf1c-d116b10e0ae8)

Reply:
However, the bigger issue is that this test approach isn't good enough, because the figure actually plots the same irrespective of the code change in this PR.  Let me get back to you with a meaningful test.

Reply:
Okay, here's a useful test:
```python
import matplotlib.pyplot as plt
from matplotlib.transforms import Affine2D

import astropy.units as u

import sunpy.map
from sunpy.data.sample import AIA_171_IMAGE, HMI_LOS_IMAGE

aia_map = sunpy.map.Map(AIA_171_IMAGE)
hmi_map = sunpy.map.Map(HMI_LOS_IMAGE)
fig = plt.figure(figsize=(16, 4))

# Panel 1: Implicit transfrom
ax1 = fig.add_subplot(1, 3, 1, projection=aia_map)
aia_map.plot(axes=ax1, clip_interval=(1, 99.99)*u.percent)
hmi_map.draw_contours([-10, 10]*u.percent)
ax1.set_title('Default, correct behavior')

# Panel 2: Explicit transform
ax2 = fig.add_subplot(1, 3, 2, projection=aia_map)
aia_map.plot(axes=ax2, clip_interval=(1, 99.99)*u.percent)
hmi_map.draw_contours([-10, 10]*u.percent, transform=ax2.get_transform(hmi_map.wcs))
ax2.set_title('Explicitly specifying the correct transform')

# Panel 3: Explicit transform with wacky rotation
ax3 = fig.add_subplot(1, 3, 3, projection=aia_map)
rotate_transform = Affine2D().rotate_deg_around(512, 512, 90)
composite_transform = rotate_transform + ax3.get_transform(hmi_map.wcs)
aia_map.plot(axes=ax3, clip_interval=(1, 99.99)*u.percent)
hmi_map.draw_contours([-10, 10]*u.percent, transform=composite_transform)
ax3.set_title('Contours rotated by 90 deg CCW')

plt.show()
```
## Before this PR
![before](https://github.com/sunpy/sunpy/assets/991759/8b90e3e3-7100-4e4f-834c-248415529de4)
## After this PR
![after](https://github.com/sunpy/sunpy/assets/991759/d3d61dbe-8ffc-4c68-8e11-22c4e4ef737d)

Reply:
@Mubin17 are you able to modify the figure test using the suggested three panel plot above?